tests/components/ruckus_unleashed/test_config_flow.py

```diff
@@ -97,3 +103,70 @@ async def test_form_unknown_error(hass):
 
     assert result2["type"] == "form"
     assert result2["errors"] == {"base": "unknown"}
+
+
+async def test_form_cannot_connect_unknown_serial(hass):
+    """Test we handle cannot connect error on invalid serial number."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == "form"
+    assert result["errors"] == {}
+
+    with patch(
+        "homeassistant.components.ruckus_unleashed.Ruckus.connect",
+        return_value=None,
+    ), patch(
+        "homeassistant.components.ruckus_unleashed.Ruckus.mesh_name",
+        return_value=DEFAULT_TITLE,
+    ), patch(
+        "homeassistant.components.ruckus_unleashed.Ruckus.system_info",
+        return_value={},
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            CONFIG,
+        )
+
+    assert result2["type"] == "form"
+    assert result2["errors"] == {"base": "cannot_connect"}
+
+
+async def test_form_duplicate_error(hass):
+    """Test we handle duplicate error."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch(
+        "homeassistant.components.ruckus_unleashed.Ruckus.connect",
+        return_value=None,
+    ), patch(
+        "homeassistant.components.ruckus_unleashed.Ruckus.mesh_name",
+        return_value=DEFAULT_TITLE,
+    ), patch(
+        "homeassistant.components.ruckus_unleashed.Ruckus.system_info",
+        return_value=DEFAULT_SYSTEM_INFO,
+    ):
+        await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            CONFIG,
+        )
+
+        future = utcnow() + timedelta(minutes=60)
+        async_fire_time_changed(hass, future)
+        await hass.async_block_till_done()
+
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": config_entries.SOURCE_USER}
+        )
+        assert result["type"] == "form"
+        assert result["errors"] == {}
+
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            CONFIG,
+        )
+
+    assert result2["type"] == "abort"
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6333224)